### PR TITLE
[sc-6705] - changed resources limits

### PR DESCRIPTION
**Pull request by maxlevinps**
low preset:
total requests low backend: 1152Mi
total limits low backend: 1968Mi

medium preset:
total requests/limits(same) medium backend: 3196Mi

original values:
total memory requests: 6582Mi
total cpu requests: 2155m

### CODE REVIEW

File: cmd/deploy.go [hunk 1]
```diff
@@ -422,25 +422,32 @@ func getChartValues(chartValues map[string]interface{}, clusterName string, depl
...
+	}
+
+	backendPresetPath, overrideBackendPreset := helm.GetBackendResourcePresetPath(allocatableResources)
+	if overrideBackendPreset {
+		overridePaths = append(overridePaths, backendPresetPath)
 	}
 
 	if viper.GetBool(LOW_RESOURCES_FLAG) {
-		resourcesTunerPresetPaths = []string{
+		overridePaths = []string{
```
[the remark this answers is not in the document]
Comment (nitzangr) [reply cmd/deploy.go:444]: Maybe skip all the logic above?

File: pkg/helm/tune.go [hunk 2]
```diff
@@ -33,34 +35,7 @@ type AllocatableResources struct {
...
-		}
-
-		presetPaths = append(presetPaths, presetPath)
-	}
-
-	return presetPaths, nil
-}
-
-func tuneAgentResourcesValues(allocatableResources *AllocatableResources) (string, error) {
+func GetAgentResourcePresetPath(allocatableResources *AllocatableResources) (string, bool) {
```
Comment (nitzangr) [inline pkg/helm/tune.go:38]: Why do we need two returned values here?

File: pkg/helm/tune.go [hunk 3]
```diff
@@ -76,13 +51,13 @@ func tuneAgentResourcesValues(allocatableResources *AllocatableResources) (strin
...
-		return "", nil
+		return NO_PRESET, false
 	}
 
-	return presetPath, nil
+	return presetPath, true
 }
 
-func tuneBackendResourcesValues(allocatableResources *AllocatableResources) (string, error) {
+func GetBackendResourcePresetPath(allocatableResources *AllocatableResources) (string, bool) {
```
Comment (nitzangr) [inline pkg/helm/tune.go:60]: Same

File: cmd/deploy.go [hunk 4]
```diff
@@ -422,25 +422,32 @@ func getChartValues(chartValues map[string]interface{}, clusterName string, depl
...
 	useExperimental := viper.GetBool(EXPERIMENTAL_FLAG)
 	if useExperimental {
 		overridePaths = append(overridePaths, EXPERIMENTAL_PRESET_PATH)
 	}
 
-	var resourcesTunerPresetPaths []string
-	if resourcesTunerPresetPaths, err = helm.GetResourcesTunerPresetPaths(deployableNodes); err != nil {
-		return nil, err
+	allocatableResources := helm.CalcAllocatableResources(deployableNodes)
+	sentryHelmContext.AllocatableResources = allocatableResources
```
Comment (nitzangr) [inline cmd/deploy.go:431]: Do we want maybe to store also `deployableNodes`?

**Review by nitzangr — CHANGES_REQUESTED** [review]

File: cmd/deploy.go (hunk 4 above)
Comment (maxlevinps) [reply cmd/deploy.go:426]: we store them

File: cmd/deploy.go [hunk 5]
```diff
@@ -427,29 +427,29 @@ func getChartValues(chartValues map[string]interface{}, clusterName string, depl
...
-	}
-
 	if viper.GetBool(LOW_RESOURCES_FLAG) {
 		overridePaths = []string{
 			helm.AGENT_LOW_RESOURCES_PATH,
 			helm.BACKEND_LOW_RESOURCES_PATH,
 		}
+	} else {
+		agentPresetPath := helm.GetAgentResourcePresetPath(allocatableResources)
+		if agentPresetPath != "" {
```
Comment (nitzangr) [inline cmd/deploy.go:440]: Use `NO_PRESET`

**Review by nitzangr — CHANGES_REQUESTED** [review]

**Review by nitzangr — APPROVED** [review]

**Pull request merged**